### ISSUE TITLE
GGClassic - Physics performance of Polygon-type causing massive framerate issues

### ISSUE THREAD

**Issue by boltactiongaming** [TheGameCreators/GameGuruRepo#5038]
Having been debugging and optimizing some old work, I found that there were significant performance bottlenecks which could be isolated to one region and one region alone: the use of the polygon collisionmode.  
This is easily duplicable.

Take an empty level.
Go to mega pack 3
scenery --> "Office & Industrial props"  (CASE MATTERS).
This version of the water cooler uses a box-type collision mesh with full transparencies, PBR, etc.

Now use any other version of that water cooler you like.  FPSC classic set, for instance.  This uses old shaders, low quality graphics, and .. the polygon collision mode.
Bump into the object while monitoring physics performance via tab screen.

You will go from very low 3-5% values on the high quality box-type one, to very high values (80%) and framerate drops on the low quality one.

I've been able to duplicate this with around thirty different models.  The fix EACH TIME is to convert them to box type.
However, not all models are suitable for this.

**Comment by boltactiongaming** [comment]
https://github.com/TheGameCreators/GameGuruRepo/issues/1140 possibly related

**Comment by MonkeyFrogStudio** [comment]
Polygon collision is the slowest, most intensive form of collision detection and should rarely, if ever, be used. It forces the engine to consider every single polygonal face for collision purposes. If it's not possible for a player to collide with, say, a small crevice or dent on an object, then there's no need at all for the engine to have to calculate collision for the polygonal faces there.

If none of the other collision methods are suitable (cylinder, sphere, box, etc.) for the shape of the object then the best option is to create your own custom collision mesh that has as few polygonal faces as possible while still approximating the shape of your object.

**Comment by davetgc** [comment]
As @MonkeyFrogStudio said, poly collision should really be avoided, as it will cause some massive slowdowns. It has it's place, but isn't recommended for general use.

**Comment by boltactiongaming** [comment]
Poly collision is the only option for models that are based on segment or building components, however, such as the asylum pieces.  That said, it feels worse than previous versions.

Obviously poly is going to be the lowest performance but we're talking a difference of practically no frame loss to a total drag down to 30 FPS with just a few models.

I don't recall it ever being that substantial or big of a drag.

**Comment by MonkeyFrogStudio** [comment]
Poly collision should not be the only options for models that are based on segments or building components. You could use Hull Decomp or Convex Hull, depending on the model. Of course, once again, the best solution is a custom collision mesh. As stated previously, polygon collision should rarely, if ever, be used.

**Comment by boltactiongaming** [comment]
Except what I'm saying is that I've tested the alternative solutions of different collision modes - they do not work, full stop.  You cannot walk through the area.  So those are non-starters as far as a resolution is concerned.

**Comment by Kitakazi** [comment]
Unless it was added at a later date I don't think Classic even has Hull Decomp. 
Boltactiongaming you should try Max man... I think you would be quite pleased. (sorry not trying to turn your bug report into a discussion just kinda shocked you're still using classic)

**Comment by MonkeyFrogStudio** [comment]
@Kitakazi Yeah, I didn't realize, when I'd posted, that this was about Classic. My bad.
